src/client.ts

```diff
@@ -165,95 +149,151 @@ export class SubscriptionClient {
   }
 
   public unsubscribe(id: number) {
-    delete this.subscriptions[id];
-    delete this.waitingSubscriptions[id];
-    let message = { id: id, type: SUBSCRIPTION_END};
-    this.sendMessage(message);
+    if (this.requests[id]) {
+      delete this.requests[id];
+    }
+    this.sendMessage(id, MessageTypes.GQL_END, undefined);
   }
 
   public unsubscribeAll() {
-    Object.keys(this.subscriptions).forEach( subId => {
-      this.unsubscribe(parseInt(subId));
+    Object.keys(this.requests).forEach( subId => {
+      this.unsubscribe(parseInt(subId, 10));
     });
   }
 
-  // send message, or queue it if connection is not open
-  private sendMessage(message: Object) {
-    switch (this.client.readyState) {
+  private executeRequest(options: RequestOptions, handler: (error: Error[], result?: any) => void): number {
+    const { query, variables, operationName } = options;
+
+    if (!query) {
+      throw new Error('Must provide a `query`.');
+    }
+
+    if (!handler) {
+      throw new Error('Must provide an `handler`.');
```

Comment:
Done regarding the backticks. However I think we should right "an hander". The "h" is a special case here.